ee/tabby-ui/app/(dashboard)/activities/components/activity.tsx

```diff
@@ -0,0 +1,333 @@
+                          <TableHead className="md:w-[25%]">Time</TableHead>
+                          <TableHead className="md:w-[25%]">Language</TableHead>
+                        </TableRow>
+                      </TableHeader>
+                      <TableBody>
+                        {userEvents?.edges
+                          .sort(
+                            (a, b) =>
+                              new Date(b.node.createdAt).getTime() -
+                              new Date(a.node.createdAt).getTime()
+                          )
+                          .map(userEvent => (
+                            <ActivityRow
+                              key={userEvent.cursor}
+                              activity={userEvent.node}
+                            />
+                          ))}
+                      </TableBody>
+                    </Table>
+                  </CardContent>
+                </>
+              )}
+            </Card>
+
+            {(data?.userEvents.pageInfo?.hasNextPage ||
+              data?.userEvents.pageInfo?.hasPreviousPage) && (
+              <div className="flex justify-end">
+                <div className="flex w-[100px] items-center justify-center text-sm font-medium">
+                  Page {page}
+                </div>
+                <div className="flex items-center space-x-2">
+                  <Button
+                    variant="outline"
+                    className="h-8 w-8 p-0"
+                    disabled={!data?.userEvents.pageInfo?.hasNextPage}
+                    onClick={e => {
+                      setQueryVariables({
+                        first: DEFAULT_PAGE_SIZE,
+                        after: data?.userEvents.pageInfo?.endCursor
+                      })
+                      setPage(page - 1)
+                    }}
+                  >
+                    <IconChevronLeft className="h-4 w-4" />
+                  </Button>
+                  <Button
+                    variant="outline"
+                    className="h-8 w-8 p-0"
+                    disabled={!data?.userEvents.pageInfo?.hasPreviousPage}
+                    onClick={e => {
+                      setQueryVariables({
+                        last: DEFAULT_PAGE_SIZE,
+                        before: data?.userEvents.pageInfo?.startCursor
+                      })
+                      setPage(page + 1)
+                    }}
+                  >
+                    <IconChevronRight className="h-4 w-4" />
+                  </Button>
+                </div>
+              </div>
+            )}
+          </main>
+        </div>
+      </div>
+    </LoadingWrapper>
+  )
+}
+
+function ActivityRow({
+  activity
+}: {
+  activity: ListUserEventsQuery['userEvents']['edges'][0]['node']
+}) {
+  const { theme } = useTheme()
+  const [members] = useAllMembers()
+  const [isCollapse, setIsCollapse] = React.useState(false)
```

Comment:
If the default value is false, semantically, should it be `isExpanded`?

Reply:
updated to using isExpanded